File: js/10-19/problem10.js
const prototypes = require('../prototypes');

// sieve of eratoshenes
var primes = function(n) {
	var answer = [];
	var sieve = function() {
		var arr = [];
		for (let i = 0; i < n; i++) {
			arr.push(true);
		}
		return arr;
	}();
	for (let p = 2; p <= n; p++) {
		if (sieve[p]) {
			answer.push(p)
			for (let j = p * p; j <= n + 1; j += p) {
				sieve[j] = false;
			}
		}
	}
	return answer;
};

console.log(primes(1999999).sum());

File: python/10-19/problem10.py
# sieve of eratosthenes
def primes(n):
    ps, sieve = [], [True] * (n + 1)
    for p in range(2, n + 1):
        if sieve[p]:
            ps.append(p)
            for i in range(p * p, n + 1, p):
                sieve[i] = False
    return ps


if __name__ == '__main__':
    print(sum(primes(2000000)))
